galley/pkg/source/kube/builtin/source.go

```diff
@@ -169,6 +169,16 @@ func (s *source) handleEvent(kind resource.EventKind, obj interface{}) {
 			return
 		}
 
+		event.Entry.Item = item
+	case resource.Deleted:
+		// Try converting the object to a protobuf message, but don't stop at failure.
+		item := s.t.ExtractResource(obj)
```

Comment:
Do we actually need this in code? If not, I'd rather have this. It will create a false sense of expectation on the client side for the availability of the resource during a delete event.

Reply:
Yes, the pod cache needs the pod IP in order to optimize its deletion from the map. If the caste fails, it has to brute force through the entire map.

Deletion events will only happen once per pod, so this is probably the least of our concerns WRT performance.